lib/core/utils/chat_notification/push_helper.dart

```diff
@@ -0,0 +1,272 @@
+
+  // Show notification
+  final person = Person(
+    name: event.senderFromMemoryOrFallback.calcDisplayname(),
+    icon:
+        avatarFile == null ? null : BitmapFilePathAndroidIcon(avatarFile.path),
+  );
+  final newMessage = Message(
+    body,
+    event.originServerTs,
+    person,
+  );
+
+  final messagingStyleInformation = Platform.isAndroid
+      ? await AndroidFlutterLocalNotificationsPlugin()
+          .getActiveNotificationMessagingStyle(id)
+      : null;
+  messagingStyleInformation?.messages?.add(newMessage);
+
+  final roomName =
+      event.room.getLocalizedDisplayname(MatrixDefaultLocalizations());
+
+  final notificationGroupId =
+      event.room.isDirectChat ? 'directChats' : 'groupChats';
+  final groupName = event.room.isDirectChat ? "Direct chat" : "Group";
+  
+  final messageRooms = AndroidNotificationChannelGroup(
+    notificationGroupId,
+    groupName,
+  );
+  final roomsChannel = AndroidNotificationChannel(
+    event.room.id,
+    roomName,
+    groupId: notificationGroupId,
+    importance: Importance.high,
+  );
+
+  await flutterLocalNotificationsPlugin
+      .resolvePlatformSpecificImplementation<
+          AndroidFlutterLocalNotificationsPlugin>()
+      ?.createNotificationChannelGroup(messageRooms);
+  await flutterLocalNotificationsPlugin
+      .resolvePlatformSpecificImplementation<
+          AndroidFlutterLocalNotificationsPlugin>()
+      ?.createNotificationChannel(roomsChannel);
+
+  final androidPlatformChannelSpecifics = AndroidNotificationDetails(
+    event.room.id,
+    roomName,
+    channelDescription: groupName,
+    number: notification.counts?.unread,
+    category: AndroidNotificationCategory.message,
+    styleInformation: messagingStyleInformation ??
+        MessagingStyleInformation(
+          person,
+          conversationTitle: roomName,
+          groupConversation: !event.room.isDirectChat,
+          messages: [newMessage],
+        ),
+    ticker: 'Unread chats',
+    // ticker: t.chat.unread_chats(unreadCount: notification.counts?.unread ?? 1),
+    importance: Importance.max,
+    priority: Priority.max,
+    groupKey: notificationGroupId,
+  );
+  const iOSPlatformChannelSpecifics = DarwinNotificationDetails();
+  final platformChannelSpecifics = NotificationDetails(
+    android: androidPlatformChannelSpecifics,
+    iOS: iOSPlatformChannelSpecifics,
+  );
+  final title = roomName;
+  Logs().i('title: $title', title);
+  Logs().i('body: $body', body);
+  await flutterLocalNotificationsPlugin.show(
+    id,
+    title,
+    body,
+    platformChannelSpecifics,
+    payload: event.roomId,
+  );
+  Logs().v('Push helper has been completed!');
+}
+
+/// Workaround for the problem that local notification IDs must be int but we
+/// sort by [roomId] which is a String. To make sure that we don't have duplicated
+/// IDs we map the [roomId] to a number and store this number.
+Future<int> mapRoomIdToInt(String roomId) async {
```

Comment:
If we need a roomId must be int, we can use `roomId.hashmap`. For each unique roomId will be a unique hashmap

Reply:
I see, good idea for improvement, this code I keep it from FluffyChat, not self-write this code, and purpose I keep this for easier to improve or monitor when FluffyChat have some stuff changed.